frontend/src/concepts/pipelines/elyra/utils.ts

```diff
@@ -63,43 +68,55 @@ export const currentlyHasPipelines = (notebook: NotebookKind): boolean =>
   !!notebook.spec.template.spec.volumes?.find((v) => v.secret?.secretName === ELYRA_SECRET_NAME);
 
 export const generateElyraSecret = (
-  dataConnectionData: AWSSecretKind['data'],
-  dataConnectionName: string,
-  namespace: string,
+  externalStorage: DSPipelineExternalStorageKind,
+  dataConnection: SecretKind,
   route: string,
-): SecretKind => ({
-  apiVersion: 'v1',
-  kind: 'Secret',
-  metadata: {
-    name: ELYRA_SECRET_NAME,
-    namespace,
-  },
-  type: 'Opaque',
-  stringData: {
-    /* eslint-disable camelcase */
-    [ELYRA_SECRET_DATA_KEY]: JSON.stringify({
-      display_name: 'Data Science Pipeline',
-      metadata: {
-        tags: [],
+): SecretKind => {
+  const { namespace } = dataConnection.metadata;
+
+  if (
+    !dataConnection.data?.[externalStorage.s3CredentialsSecret.accessKey] ||
+    !dataConnection.data?.[externalStorage.s3CredentialsSecret.secretKey]
+  ) {
+    throw new Error(
+      'Error generating elyra secret, please check the DataSciencePipelinesApplication CR.',
+    );
+  }
```

Comment:
Not sure if there is a better way besides simply throwing the error here. @andrewballantyne Any thoughts on how we should deal with the error? If I simply throw it here, it will show the page as I put in the description.

Reply:
Throwing it gets ignored in the calling try/catch -- sooooo... this is fine, but not used?

Reply:
Yeah it's used actually... See the description of this PR, this is the result. I don't know if there is a better way to catch it and deal with it.

Reply:
We could do a push notification like we do in model serving (if i recall correctly). Only show on error.

Reply:
resolved